bindings/csharp: pluto example

Adrian-Stanea opened the pull request:
- How to use C# bindings to control libiio devices.
- Configure Pluto and use RX and TX interfaces to send and receive data.
- Nice to have for EZ questions such as:  https://ez.analog.com/adieducation/university-program/f/q-a/574101/adalam-pluto-sdr-via-c

Adrian-Stanea committed:
bindings/csharp: pluto example

- How to use C# bindings to control libiio devices.
- Configure Pluto and use RX and TX interfaces to send and receive data.
- Nice to have for EZ questions such as:  https://ez.analog.com/adieducation/university-program/f/q-a/574101/adalam-pluto-sdr-via-c
- Ported to libiio v1 API (Stream/Block, ChannelsMask, attrs.Find, etc)

Signed-off-by: Adrian Stanea <Adrian.Stanea@analog.com>
Co-Authored-By: Marius Lucacel <marius.lucacel@analog.com>

rgetz commented:
should fix the codacy issues first : 
https://app.codacy.com/gh/analogdevicesinc/libiio/commit?cid=775805355&fileBranchId=38255329#newIssuesView

-Robin

rgetz commented:
IMHO - To make sure it doesn't bit rot, you should build in in CI.

Would defer to @pcercuei to see if he wants to do that before going through the effort though...

pcercuei commented on bindings/csharp/examples/PlutoExample.cs:
```diff
@@ -0,0 +1,147 @@
+    static class PlutoExample
+    {
+        const int INT16_BYTE_STEP = 2;
+        const string URI = "ip:192.168.2.1";
```
Should probably use `ip:pluto.local` instead.

pcercuei commented on bindings/csharp/examples/PlutoExample.cs:
```diff
@@ -0,0 +1,147 @@
+        const string RXBW = TXBW;
+        const string RXFS = TXFS;
+        const string GAIN_CTRL_MODE = "slow_attack";
+        const string HW_GAIN = "-30";
```
Why not use a `double` value? Same for TXLO / TXBW / etc. above.

999Marius commented:
**Update v1:**
- Ported to libiio v1 API (Stream/Block, ChannelsMask, attrs.Find)
- Tested on real Pluto

Note: force-pushed the rebased branch before retargeting the PR to main, which temporarily showed the entire main history in the diff, but it has been fixed:)

999Marius committed:
bindings/csharp: build examples in CI

Add CMake targets to compile C# examples as part of the build.
Examples are build only, no execution, as they require real hardware.

Signed-off-by: Marius Lucacel <marius.lucacel@analog.com>

999Marius commented:
**Update v2:**
- Made all csharp/example to be built in CI

999Marius commented:
**Update v2:**
  - Moved examples CMake into a dedicated `bindings/csharp/examples/CMakeLists.txt`

999Marius commented:
**Update v3:**
- Used the new Dictionary look up for attributes
- Added a try catch to the Context creation

dNechita reviewed (approved):
Looks good!
